### ISSUE TITLE
Environment broken (container restarting) in last version

### ISSUE THREAD

**Issue by rpcarleto** [docker-library/mysql#1056]
I have a docker compose that provides mysql container, but when pull new version, it has been failed.
I try to back to preview version, but downgrade is not possible.

 db_mysql:
    image: mysql:8
    command:
      - --mysql-native-password=ON
      - --log-error-verbosity=3
    volumes:
            #- ./sql/onloc.sql:/docker-entrypoint-initdb.d/onloc.sql
            #- ./sql/mycustom.cnf:/etc/mysql/conf.d/custom.cnf
      - ./db-volume:/var/lib/mysql
      - ./db-logs:/var/log/mysql
    environment:
      - TZ=America/Sao_Paulo
      - MYSQL_ROOT_PASSWORD=XXXXXXX

Container keep restarting with errors

2024-05-07T16:15:21.374619Z 0 [Note] [MY-013667] [Server] Error-log destination "stderr" is not a file. Can not restore error log messages from previous run.
2024-05-07T16:15:21.103149Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-05-07T16:15:21.372181Z 0 [Note] [MY-013932] [Server] BuildID[sha1]=353ae40c5329277f3033302aee25ef64367c8a66
2024-05-07T16:15:21.372187Z 0 [Note] [MY-010949] [Server] Basedir set to /usr/.
2024-05-07T16:15:21.372200Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.4.0) starting as process 1
2024-05-07T16:15:21.377189Z 0 [Note] [MY-012366] [InnoDB] Using Linux native AIO
2024-05-07T16:15:21.377363Z 0 [Note] [MY-015140] [InnoDB] Using innodb_flush_method=O_DIRECT
2024-05-07T16:15:21.377706Z 0 [Note] [MY-010747] [Server] Plugin 'FEDERATED' is disabled.
2024-05-07T16:15:21.377866Z 0 [Note] [MY-010747] [Server] Plugin 'ndbcluster' is disabled.
2024-05-07T16:15:21.377954Z 0 [Note] [MY-010747] [Server] Plugin 'ndbinfo' is disabled.
2024-05-07T16:15:21.378042Z 0 [Note] [MY-010747] [Server] Plugin 'ndb_transid_mysql_connection_map' is disabled.
2024-05-07T16:15:21.379267Z 0 [Warning] [MY-010001] [Server] Can't create thread to handle bootstrap (errno: 1)
2024-05-07T16:15:21.379370Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2024-05-07T16:15:21.379468Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-05-07T16:15:21.379563Z 0 [Note] [MY-010120] [Server] Binlog end
2024-05-07T16:15:21.379702Z 0 [Note] [MY-015019] [Server] MySQL Server: Plugins Shutdown - start.
2024-05-07T16:15:21.379776Z 0 [Note] [MY-010733] [Server] Shutting down plugin 'MyISAM'
2024-05-07T16:15:21.379866Z 0 [Note] [MY-010733] [Server] Shutting down plugin 'InnoDB'
2024-05-07T16:15:21.379949Z 0 [Note] [MY-010733] [Server] Shutting down plugin 'CSV'
2024-05-07T16:15:21.380021Z 0 [Note] [MY-010733] [Server] Shutting down plugin 'daemon_keyring_proxy_plugin'
2024-05-07T16:15:21.380164Z 0 [Note] [MY-015020] [Server] MySQL Server: Plugins Shutdown - end.
2024-05-07T16:15:21.380515Z 0 [Note] [MY-015021] [Server] MySQL Server: Components Shutdown - start.
2024-05-07T16:15:21.381187Z 0 [Note] [MY-015022] [Server] MySQL Server: Components Shutdown - end (with return value = 0).
2024-05-07T16:15:21.381368Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.4.0)  MySQL Community Server - GPL.
2024-05-07T16:15:21.381554Z 0 [System] [MY-015016] [Server] MySQL Server - end.

**Comment by LaurentGoderre** [comment]
Looks like a duplicate of #1048

**Comment by rpcarleto** [comment]
No, it's not. The error message is errors is generic and I know about the flag --mysql-native-password=ON

Trying to understand whats failed, yet.

**Comment by rpcarleto** [comment]
I don't know if helps, but swap to tag "oraclelinux8" works as expected. I would like to know why works on this tag but 8.4 doesn't.

**Comment by tianon** [comment]
Ah, that's probably #1055

**Comment by yosifkit** [comment]
Or perhaps this is a duplicate of #1058, so https://github.com/docker-library/mysql/issues/1058#issuecomment-2105000516 might be a fix.
